File: src/test/java/challenges/c20201001_20201007/NumberOfRecentCalls/NumberOfRecentCallsTest.java
package challenges.c20201001to20201007.NumberOfRecentCalls;

import static org.junit.jupiter.api.Assertions.assertEquals;

import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Story;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Tags;
import org.junit.jupiter.api.Test;

@Tags({@Tag("tag Java")})
@Epic("epic Challenges")
@Feature("feature 20201001-20201007")
@Story("story Number of recent calls")
@DisplayName("suite Java")
@SuppressWarnings("PMD")
public class NumberOfRecentCallsTest {
    @Test
    @Tags({@Tag("tag Java")})
    @DisplayName("test Number of recent calls")
    void testNumberOfRecentCalls() {
        final RecentCounter recentCounter = new RecentCounter();
        assertEquals(1, recentCounter.ping(1));
        assertEquals(2, recentCounter.ping(100));
        assertEquals(3, recentCounter.ping(3001));
        assertEquals(3, recentCounter.ping(3002));
    }
}

File: src/test/java/challenges/c20201001_20201007/NumberOfRecentCalls/NumberOfRecentCallsTest.cs
// <copyright file="NumberOfRecentCallsTest.cs" company="PlaceholderCompany">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>

namespace Challenges.C20201001_20201007.NumberOfRecentCalls
{
    using System.Linq;
    using NUnit.Allure.Core;
    using NUnit.Framework;
    using NUnit.Allure.Attributes;
    using NUnit.Allure.Core;
    using NUnit.Framework;

    [TestFixture]
    [AllureNUnit]
    [AllureEpic("epic Challenges")]
    [AllureFeature("feature 20201001-20201007")]
    [AllureStory("story Recent counter")]
    [AllureSuite("suite C#")]
    [AllureTag("tag C#")]
    public class RecentCounterTest
    {
        [SetUp]
        public void Setup()
        {
        }

        [Test]
        [AllureName("test Recent counter")]
        public void PingTest()
        {
            RecentCounter obj = new RecentCounter();
            var calls = new int[] { 1, 100, 3001, 3002 };
            calls.ToList().ForEach(call => obj.Ping(call));
            /*
            int t = 200;
            int param_1 = obj.Ping(t);
            */
        }
    }
}